openedx/core/djangoapps/user_authn/views/logout.py

```diff
@@ -55,8 +57,19 @@ def dispatch(self, request, *args, **kwargs):
         # Get the list of authorized clients before we clear the session.
         self.oauth_client_ids = request.session.get(edx_oauth2_provider.constants.AUTHORIZED_CLIENTS_SESSION_KEY, [])
 
+        user = copy.copy(request.user)
```

Comment:
Tests are failing, this line needs to be changed. Maybe we need to check if user is authenticated.

Reply:
yeah looking into this, I am not sure if it's breaking on this